### PR TITLE
Refactor/fullname pipe

### PR DESCRIPTION
**Pull request by jnizet**
on top of #59

### CODE REVIEW

File: frontend/src/app/fullname.pipe.spec.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+  const pipe = new FullnamePipe();
+
+  it('should not display anything if no first name or lastname of nickname', () => {
+    const person : PersonModel = {} as PersonModel
+    expect(pipe.transform(person)).toBe('');
+  });
+
+  it('should display first name and lastname when no nick name', () => {
+    const person : PersonModel = {
```
Comment (cexbrayat) [inline frontend/src/app/fullname.pipe.spec.ts]: I saw you use this in several places: `const myVar: MyType = {} as MyType`. I did not checked but I think the type inference allows `const myVar = {} as MyType` right?

File: frontend/src/app/fullname.pipe.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+import { PersonModel } from './models/person.model';
+
+@Pipe({
+  name: 'fullname'
+})
+export class FullnamePipe implements PipeTransform {
+
+  transform(person: PersonModel): string {
+    const elements = [person.firstName, person.lastName, person.nickName ? `(${person.nickName})` : null];
+    return elements.filter(e => e).join(' ');
```
Comment (cexbrayat) [inline frontend/src/app/fullname.pipe.ts]: Funny to do it this way :) Took me a few seconds to understand what was going on. 

I would have simply returned: 

```typescript
const nickName = person.nickName ? ` (${person.nickName})` : '';
return `${person.firstName} ${person.lastName}${nickname}`;
```

Less pretty maybe.

File: frontend/src/app/fullname.pipe.spec.ts (hunk 1 above)
Comment (jnizet) [inline frontend/src/app/fullname.pipe.spec.ts]: You're absolutely right.

File: frontend/src/app/fullname.pipe.ts (hunk 2 above)
Comment (jnizet) [inline frontend/src/app/fullname.pipe.ts]: The idea is to eliminate null values. Your code would print 'null null' if the first name and the last name are null.

**Pull request merged**